packages/common/sfnStep.js

```diff
@@ -5,14 +5,48 @@ const { pullStepFunctionEvent } = require('./aws');
 const log = require('./log');
 const StepFunctions = require('./StepFunctions');
 
+const taskExitedEventType = 'TaskStateExited';
+const taskExitedEventDetailsKey = 'stateExitedEventDetails';
+
+const getStepExitedEvent = (events, lastStepEvent) =>
+  events.find(
+    (event) =>
+      event.type === taskExitedEventType
+        && event.previousEventId === lastStepEvent.id
+  );
+
+const getTaskExitedEventOutput = (event) => event[taskExitedEventDetailsKey].output;
+
 /**
  * `SfnStep` provides methods for getting the output of a step within an AWS
  * Step Function for a specific execution.
 */
 class SfnStep {
-  constructor() {
-    this.taskExitedEvent = 'TaskStateExited';
-    this.taskExitedDetailsKey = 'stateExitedEventDetails';
+  /**
+   * Parse the step message.
+   *
+   * Merge possible keys from the CMA in the input and handle remote message
+   * retrieval if necessary.
+   *
+   * @param {Object} stepMessage - Details for the step
+   * @param {Object} stepMessage.input - Object containing input to the step
+   * @param {string} [stepName] - Name for the step being parsed. Optional.
+   * @returns {Object} - Parsed step input object
+   */
+  static async parseStepMessage(stepMessage, stepName) {
+    let parsedStepMessage = stepMessage;
+    if (stepMessage.cma) {
+      parsedStepMessage = { ...stepMessage, ...stepMessage.cma, ...stepMessage.cma.event };
+      delete parsedStepMessage.cma;
+      delete parsedStepMessage.event;
+    }
+
+    if (parsedStepMessage.replace) {
```

Comment:
This is new: previously it checked `stepMessage.replace` and not `parsedStepMessage.replace`. However, my understanding is that the above code for massaging the event is for backwards compatibility with older CMA versions where the step message may actually have `cma` or `cma.event` keys. If that were the case, the`replace` key may exist on `cma.event.replace`, meaning we want to check `parsedStepMessage.replace` to see if remote message replacement is necessary.

If I'm missing something or totally misunderstanding, please let me know though